### DISCUSSION TITLE
A3 - living on the edge

### DISCUSSION THREAD

**Discussion by myovchev** [apostrophecms/apostrophe#2708] [General]
Not really a discussion :) 
I always end up checking the right syntax to install a GitHub repository as npm package. So I'll just drop the appropriate way to add Apostrophe v3 pointing directly to the current development branch for any DEVELOPMENT reasons (please, don't even try it in production).

Find the line in your package.json saying something like:
```json
...
"apostrophe": "apostrophe@^3.0.0-alpha.3",
...
```
change it to:
```json
...
"apostrophe": "git://github.com/apostrophecms/apostrophe.git#3.0",
...
```
and do `npm i`

Another way to do it is via command line:
```sh
# if already available as dependency
npm uninstall apostrophe
# add the 3.0 branch as package
npm install git://github.com/apostrophecms/apostrophe.git#3.0
```
and check the package.json (should look like the example above)

The above (the package.json part) works as expected in Lerna projects as well.

**Comment by boutell** [comment]
True! This also works and it's a bit more economical:

npm install "github:apostrophecms/apostrophe#3.0"


On Thu, Feb 4, 2021 at 9:50 AM Miro Yovchev <notifications@github.com>
wrote:

> Not really a discussion :)
> I always end up checking the right syntax to install a GitHub repository
> as npm package. So I'll just drop the appropriate way to add Apostrophe v3
> pointing directly to the current development branch for any DEVELOPMENT
> reasons (please, don't even try it in production).
>
> Find the line in your package.json saying something like:
>
> ..."apostrophe": "apostrophe@^3.0.0-alpha.3",
> ...
>
> change it to:
>
> ..."apostrophe": "git://github.com/apostrophecms/apostrophe.git#3.0",
> ...
>
> and do npm i
>
> Another way to do it is via command line:
>
> # if already available as dependency
> npm uninstall apostrophe# add the 3.0 branch as package
> npm install git://github.com/apostrophecms/apostrophe.git#3.0
>
> and check the package.json (should look like the example above)
>
> The above (the package.json part) works as expected in Lerna projects as
> well.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/2708>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27PBHESD6I3LE3ITAD3S5KXZTANCNFSM4XC6YANA>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by myovchev** [reply]
I've missed that direct github support - could have something to do with Microsoft owning both github and npm :) (I guess it's older than that but still good joke!)
